py_ecc/optimized_bn128/optimized_curve.py

```diff
@@ -142,15 +142,13 @@ def normalize(pt: Optimized_Point3D[Optimized_Field]) -> Optimized_Point2D[Optim
 
 # Convert P => -P
 def neg(pt: Optimized_Point3D[Optimized_Field]) -> Optimized_Point3D[Optimized_Field]:
-    if pt is None:
-        return None
+    if is_inf(pt):
+        return pt
```

Comment:
If I'm doing the profiling right, is_inf check is no faster than just do the calculation
https://gist.github.com/ChihChengLiang/ac2a89f3ff90cb0c559f1d2423f5aa1c#file-profile_inf_check-py

Reply:
Makes sense to just remove the check entirely, as done in `twist`

PS: thanks for showing your work RE profiling

Reply:
Sorry, the previous profiling is incorrect for not using real `Z1` and `is_inf`. I did a new one, and the conclusion still holds. no-check is faster.
https://gist.github.com/ChihChengLiang/ac2a89f3ff90cb0c559f1d2423f5aa1c#file-fixed_profile_inf_check-py